spec/tbd_tests_spec.rb

```diff
@@ -1237,6 +1239,31 @@
     expect(model.getSurfacePropertyExposedFoundationPerimeters.size).to eq(1)
     expect(model.getFoundationKivas.size).to eq(1) # !4 ... previously purged
 
+    # By default, KIVA foundation objects have a 200mm 'wall height above grade'
+    # value, i.e. a top, 8-in section exposed to outdoor air. This seems to
+    # generate the following EnergyPlus warning:
+    #
+    #   ** Warning ** BuildingSurface:Detailed="OPENAREA 1 WALL 5", Sun Exposure="SUNEXPOSED".
+    #   **   ~~~   ** ..This surface is not exposed to External Environment.  Sun exposure has no effect.
+    #
+    # Initial attempts to get rid of the warning include resetting both wind and
+    # sun exposure AFTER setting boundary conditions to "Foundation", e.g.
+    #
+    #   expect(oa15.setOutsideBoundaryCondition("Foundation")).to be true
+    #   expect(oa15.setWindExposure("NoWind")).to be true
+    #   expect(oa15.setSunExposure("NoSun")).to be true
+    #
+    # Alas, both "exposures" can reset in the saved OSM. One solution is to
+    # first set the 'wall height above grade' value to 0:
+    kf = model.getFoundationKivas.first
+    expect(kf.isWallHeightAboveGradeDefaulted).to be true
+    expect(kf.wallHeightAboveGrade.round(1)).to eq(0.2)
+    expect(kf.setWallHeightAboveGrade(0)).to be true
+    expect(kf.isWallHeightAboveGradeDefaulted).to be false
+    expect(kf.wallHeightAboveGrade.round).to eq(0)
+    expect(oa15.setWindExposure("NoWind")).to be true
+    expect(oa15.setSunExposure("NoSun")).to be true
```

Comment:
By default, KIVA foundation objects have a 200mm _wall height above grade_ value, i.e. a top, 8-in section exposed to outdoor air. This seems to generate the following EnergyPlus warning:

```
** Warning ** BuildingSurface:Detailed="OPENAREA 1 WALL 5", Sun Exposure="SUNEXPOSED".
**   ~~~   ** ..This surface is not exposed to External Environment.  Sun exposure has no effect.
```

Initial attempts to get rid of the warning include resetting both wind and sun exposure AFTER setting boundary conditions to "Foundation", e.g.

```
expect(oa15.setOutsideBoundaryCondition("Foundation")).to be true
expect(oa15.setWindExposure("NoWind")).to be true
expect(oa15.setSunExposure("NoSun")).to be true
```

Alas, both "exposures" end up being reset in the saved OSM. One solution is to first set the _wall height above grade_ value to 0. Works. More tests needed (to come).